backend/capellacollab/core/authentication/routes.py

```diff
@@ -0,0 +1,158 @@
+# SPDX-FileCopyrightText: Copyright DB InfraGO AG and contributors
+# SPDX-License-Identifier: Apache-2.0
+
+import typing as t
+
+import fastapi
+from sqlalchemy import orm
+
+from capellacollab.config import config
+from capellacollab.core import database
+from capellacollab.core.authentication import injectables as auth_injectables
+from capellacollab.users import crud as users_crud
+from capellacollab.users import exceptions as users_exceptions
+from capellacollab.users import models as users_models
+
+from . import api_key_cookie, exceptions, flow, models
+
+router = fastapi.APIRouter()
+
+
+@router.get("", name="Get redirect URL for Auth Server")
+async def get_redirect_url(response: fastapi.Response) -> str:
+    authorization_url, state = flow.get_authorization_url_with_state()
+
+    delete_access_token_cookies(response)
+    delete_secure_cookie(response, "refresh_token", "/api/v1")
+
+    set_secure_cookie(response, "state", state, "/api/v1")
+
+    return f"{authorization_url}?state={state}"
+
+
+@router.post("/tokens", name="Create access_token")
+async def api_get_token(
+    token_request: models.TokenRequest,
+    response: fastapi.Response,
+    state: t.Annotated[str | None, fastapi.Cookie()] = None,
+    db: orm.Session = fastapi.Depends(database.get_db),
+):
+    delete_secure_cookie(response, "state", "/api/v1")
+
+    if not state:
+        raise exceptions.NoStateProvidedError()
+
+    if token_request.state != state:
+        raise exceptions.StateMismatchError()
+
+    tokens = flow.get_access_token(token_request.code)
+
+    validate_tokens_and_update_cookies(
+        response, db, tokens["access_token"], tokens["refresh_token"]
+    )
+
+    return True
+
+
+@router.put("/tokens", name="Refresh the access_token")
```

Comment:
Instead of a separate route, can't we just automatically refresh the token when the access token is expired? If that would be the case, we could get rid of the auth handling in the frontend.

Reply:
Yes, that should definitely be possible, but I am not sure how much effort it would take to change that. Do you want to include this in this PR or should we do it in a follow up?

Reply:
Fine for now since it works - can be improved as follow-up.